torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -151,14 +153,16 @@ def join(self, rank, data):
                     self._start_cond.notify()
 
             if rank == 0:
-                while self._count < self._world_size:
-                    self._start_cond.wait()
+                self._start_cond.wait_for(lambda: self._count == self._world_size or self._pg._terminate.is_set())
```

Comment:
if you already did the `notify` for start and end cond, we shouldn't wait for the terminate event right?